pandas/tests/groupby/test_groupby.py

```diff
@@ -652,6 +652,8 @@ def test_groupby_as_index_select_column_sum_empty_df():
     left = df.groupby(by="A", as_index=False)["B"].sum(numeric_only=False)
 
     expected = DataFrame(columns=df.columns[:2], index=range(0))
+    # GH#?? - Columns after selection shouldn't retain names
+    expected.columns.names = [None]
```

Comment:
This is a change in behavior; currently `.groupby(..., as_index=False)["B"]` retains the index name. We could still preserve the index name here, but it'd be a bit of a hack to add on. By not preserving the name, we make `df["A"].groupby(..., as_index=False)` behave the same as `df.groupby(..., as_index=False)["A"]`, which I think is desirable.

But would appreciate others thoughts here.

Reply:
Hmm I suppose in theory it would be nice if both operations kept the column name since the result always contains portion of the old column object?

But in this case I suppose it's minor and there's an argument that a groupby result produces a "new column"